### DISCUSSION TITLE
Using Nuxt with Firebase - No Firebase App '[DEFAULT]' has been created

### DISCUSSION THREAD

**Discussion by m6keller** [nuxt/nuxt#17641] [Questions]
I get this error when using I try to use the getAuth() function from "firebase/auth" in my Nuxt 3 project:
```
Firebase: No Firebase App '[DEFAULT]' has been created - call Firebase App.initializeApp() (app/no-app).
```

I have looked at other GitHub discussions that have said to make sure that initializeApp() must be used before using other functions, but this is true in my code. Below is my function that causes the error:

```
export default defineNuxtPlugin( (nuxtApp) => {
    
    const config = useRuntimeConfig();
    
    const firebaseConfig = {
        apiKey: config.FIREBASE_API_KEY
      };
    
    // Initialize Firebase
    initializeApp(firebaseConfig);

    initUser();
  
    const auth = getAuth();
  
    nuxtApp.vueApp.provide('auth', auth);
    nuxtApp.provide('auth', auth);  
});
```

When I comment out each line after `initUser();`, my application works (without authentication). It would seem that my firebase app should already have been intialized when the `getAuth()` function is called and I am unsure if this is a bug or a mistake in my program.

Thank you

**Comment by DevJoghurt** [comment]
It seems that you are not initializing firebase properly.
If you call "initializeApp" you get an instance of your firebase app. You have to insert it into the "getAuth" function.
It is also important that your firebase config is complete.

```
export default defineNuxtPlugin( (nuxtApp) => {
    
    const config = useRuntimeConfig();
    
    const firebaseConfig = {
        projectId: "XXXXXX",
        appId: "XXXXX",
        apiKey: config.FIREBASE_API_KEY
      };
    
    // Initialize Firebase
    const firebaseApp = initializeApp(firebaseConfig);

    initUser();
  
    const auth = getAuth(firebaseApp);
  
    nuxtApp.vueApp.provide('auth', auth);
    nuxtApp.provide('auth', auth);  
});
```

**Reply by m6keller** [reply]
Thank you
